### ISSUE TITLE
Results on PC and on Android are very different

### ISSUE THREAD

**Issue by NazaRik555** [tensorflow/tensorflow#74312]
Hi everyone!
I'm using two models, for detection and then classification
Detector YOLOv8n, and it's okay with it
But classifier - EfficientNetB0 has some troubles

Here is how I'm running model on PC:
```
import numpy as np
import tensorflow as tf

# Load the TFLite model and allocate tensors.
interpreter = tf.lite.Interpreter(model_path="classifier_best.tflite")
interpreter.allocate_tensors()

# Get input and output tensors.
input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()

input_data = tens.detach().numpy()
# Test the model on random input data.
#input_shape = input_details[0]['shape']
#input_data = np.array(np.random.random_sample(input_shape), dtype=np.float32)
interpreter.set_tensor(input_details[0]['index'], input_data)

interpreter.invoke()

# The function `get_tensor()` returns a copy of the tensor data.
# Use `tensor()` in order to get a pointer to the tensor.
output_data = interpreter.get_tensor(output_details[0]['index'])
print(np.argmax(output_data))
```

and it's okay, but predictions on Android on the same image is too different

I'm running model on Android with cropped image from YOLO detector. Here's my cropping code:
```
val frame_x1 = it.x1*frame.width
            val frame_y1 = it.y1*frame.height
            val frame_w = it.w*frame.width
            val frame_h = it.h*frame.height
            try{
                val tmpBitmap = Bitmap.createBitmap(frame, frame_x1.toInt()-5, frame_y1.toInt()-5, frame_w.toInt()+5, frame_h.toInt()+5)
                classifier?.invoke(tmpBitmap)
                tmpBitmap.compress(Bitmap.CompressFormat.JPEG, 100, fos)
            } catch (e: Exception){
                Log.d("183", ""+e)
            }
```
And here is my invoke fun in classifier class:
```
fun invoke(frame: Bitmap) {
        if (tensorWidth == 0) return
        if (tensorHeight == 0) return

        var inferenceTime = SystemClock.uptimeMillis()

        val resizedBitmap = Bitmap.createScaledBitmap(frame, tensorWidth, tensorHeight, false)

        val tensorImage = TensorImage(INPUT_IMAGE_TYPE)
        tensorImage.load(resizedBitmap)
        val processedImage = imageProcessor.process(tensorImage)
        val imageBuffer = processedImage.buffer

        val output = TensorBuffer.createFixedSize(intArrayOf(1 , numClass) , OUTPUT_IMAGE_TYPE)
        interpreter.run(imageBuffer, output.buffer)

        val outputArray = output.floatArray

        val predictions = mutableListOf<Prediction>()

        outputArray.forEachIndexed { index, float ->
            if (float > CONFIDENCE_THRESHOLD) {
                predictions.add(
                    Prediction(
                        id = index,
                        name = labels[index],
                        score = float
                    )
                )
            }
        }
```
Am I doing something wrong?

**Comment by tilakrayal** [comment]
@NazaRik555,
Could you please provide the tensorflow version you are using to test the above mentioned code? Thank you!

**Comment by NazaRik555** [comment]
> @NazaRik555, Could you please provide the tensorflow version you are using to test the above mentioned code? Thank you!

PC - 2.17.0


Android:

```
    implementation("org.tensorflow:tensorflow-lite:2.16.1")
    implementation("org.tensorflow:tensorflow-lite-support:0.4.4")

    implementation("org.tensorflow:tensorflow-lite-gpu-delegate-plugin:0.4.4")
    implementation("org.tensorflow:tensorflow-lite-gpu-api:2.16.1")
    implementation("org.tensorflow:tensorflow-lite-api:2.16.1")
    implementation("org.tensorflow:tensorflow-lite-gpu:2.16.1")
    implementation("org.tensorflow:tensorflow-lite-select-tf-ops:2.16.1")

    implementation("org.tensorflow:tensorflow-lite-metadata:0.4.4")
```

**Comment by gaikwadrahul8** [comment]
Hi, @NazaRik555 

I apologize for the delayed response and thank you for bringing this issue to our attention, if possible could you please help me with Github repo with minimal code to reproduce the same behavior from our end with complete steps and as much as details which will help us to investigate this issue further ?

If you followed any TFLite official documentation please help us with that to follow the exact steps to reproduce the same behavior from our end.

Thank you for your cooperation and patience.

**Comment by NazaRik555** [comment]
> Hi, @NazaRik555
> 
> I apologize for the delayed response and thank you for bringing this issue to our attention, if possible could you please help me with Github repo with minimal code to reproduce the same behavior from our end with complete steps and as much as details which will help us to investigate this issue further ?
> 
> If you followed any TFLite official documentation please help us with that to follow the exact steps to reproduce the same behavior from our end.
> 
> Thank you for your cooperation and patience.

You need only android project or my pytorch and convertation codes too?

**Comment by NazaRik555** [comment]
Today I noticed strange thing:
Android inference gives me 1.0 or 0.0 from last Softmax layout.
Maybe it somehow connected with bad results?

**Comment by NazaRik555** [comment]
@gaikwadrahul8 

https://github.com/NazaRik555/CarDetectionClassification
Here's android project

And here's models for project
detector: https://drive.google.com/file/d/1I1A7gchVQnGh5-yMfkqYKVYVzUTh8SdW/view?usp=drivesdk
classifier: https://drive.google.com/file/d/1hy4soyldC9ZiMpcmfxYfMOLC3WS8yxSm/view?usp=drivesdk

**Comment by NazaRik555** [comment]
@gaikwadrahul8 
Do u have some ideas I can try?

**Comment by gaikwadrahul8** [comment]
Hi, @NazaRik555 

I apologize for the delayed response, may I know on which OS have you tried above experiment ? Thank you

**Comment by NazaRik555** [comment]
@gaikwadrahul8 

Windows OS
And if it can help, without Softmax activation I have about 2-5 max score on PC and about 20k max score on android

**Comment by gaikwadrahul8** [comment]
Hi, @NazaRik555

I apologize for the delayed response and It seems like your invitation request got expired I really sorry I missed that one if possible could you please send me once again ? Thank you.

**Comment by NazaRik555** [comment]
Hi, @gaikwadrahul8 
Sorry for late, could you, please, re-open this thread? And I sent invitation again

**Comment by gaikwadrahul8** [comment]
Hi, @NazaRik555 

Thank you for sending me invitation again and now I'm able to access your Github repo, I've re-opened this issue and will try to replicate the similar behavior from my end.

Thank you for your cooperation and patience.

**Comment by gaikwadrahul8** [comment]
Hi, @NazaRik555
I apologize for the delayed response, I was trying to replicate the similar behavior from my end after downloading both your models and keeping them into Assets folder and updated model name correctly but unfortunately I'm unable to replicate the issue with emulator device may be you're trying on actual android physical device 

Hi, @pkgoogle 
Could you please take a look into this issue. Thank you.

**Comment by pkgoogle** [comment]
Hi @NazaRik555, your android project link appears to be broken/404 now. If you can share your exported android project, that will help immensely. Thank you for your help.

**Comment by NazaRik555** [comment]
@pkgoogle https://github.com/NazaRik555/CarDetectionClassification and I've sent u an invitation to contributors

**Comment by pkgoogle** [comment]
Hi @NazaRik555, I don't see you cropping the image on "PC", if you are please do share that code as well. If you are not, then that's possibly the source of the discrepancy.

**Comment by NazaRik555** [comment]
Hi @pkgoogle, I've tried to run classifier at start of the android app without detector so without cropping. I've used same image, here is the link for it https://imgur.com/Dz9BZnk
Here is the code of loading img from assets and getting results in CarDetector.kt during initializing
```
val ims = context.assets.open("solara_cropped.jpg")
val bmp = BitmapFactory.decodeStream(ims)

classifier?.invoke(bmp)
```

Here is my onResult function:
```
override fun onResult(data: List<Prediction>, inferenceTime: Long) {
        Log.d("clf_result", data[0].name)
        try {
            clfResults.add(data[0].name+" "+plateResult)
        } catch (e: Exception){
            clfResults.add(data[0].name)
        }

        classifierConfs.add(data[0].score)
        classifierTime += inferenceTime
    }
```

**Comment by pkgoogle** [comment]
Hi @NazaRik555, I was able to get it running but I couldn't invoke your invoke function at least on emulator. Can you upload your modified code to your github repo (you can put it to a different branch if you prefer). Perhaps that will be easiest, also if it's not available anywhere, the "solara_cropped.jpg" file.
